course6/toyapp_alexs/displaylib/vector.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include <array>
+
+namespace Display
+{
+	template<class T, std::size_t S>
+	struct Vector
+	{
+	};
+
+	template<>
+	struct Vector<int, 2>
+	{
+		int x;
+		int y;
+	};
+
+	template<>
+	struct Vector<double, 3>
```

Comment:
This is technically correct, but rather pointless.
The idea of templates is to allow user to use own types/sizes.

I was expecting something more like:
```
template<typename T, int S>
class Vector
{
    std::array<T, S> data;
public:
   // ...
};
```